pkg/block/local/walker.go

```diff
@@ -81,7 +83,7 @@ func (l *LocalWalker) Walk(_ context.Context, storageURI *url.URL, options WalkO
 			}
 			const dirPerm = 0o755
 			_ = os.MkdirAll(l.cacheLocation, dirPerm)
-			cacheName := filepath.Join(l.cacheLocation, nanoid.Must()+"-import.json")
+			cacheName := filepath.Join(l.cacheLocation, gonanoid.Must()+"-import.json")
 			const cachePerm = 0o644
 			if err := os.WriteFile(cacheName, jsonData, cachePerm); err != nil {
```

Comment:
What deletes this file?  Especially consider what happens during failure: if the file is not deleted, the user will try again... and _nothing_ will release this file.  Especially especially consider what happens during _unexpected_ failure.

The usual way to handle temporary files in POSIX is:
1. Create a temporary file (`creat`).
1. Delete it.
1. Write whatever contents you need.
1. `lseek` back to the start of the file.
1. Read whatever you need.
1. ...
1. `close` the file.

POSIX filesystems _explicitly_ support this mode: A deleted file has no name, and is deleted when no open file descriptors remain on it.  This pattern is already [implemented](https://github.com/treeverse/lakeFS/blob/03df0f32810c4e74f736278c6cbc4fa63b75d082/pkg/fileutil/writer_reader.go#L32-L41) in lakeFS.

Reply:
In case a continuation token (file) is required for the next read, we cannot deleted the file.
The write and read will happen in a different context, and therefore we cannot use the pattern you described.
I added a remove in case of an error.
I agree we do need to have a cleanup mechanism for stale continuation token files, but not quite sure how we should do it - in any case I don't think this should be under the scope of this PR.
I'll open a new issue for it.

Reply:
Understood that this is not part of this PR.  Note also that keeping temporary files in the permanent storage area (previous comment) makes this even more confusing.